internal/fs/walk.go

```diff
@@ -27,15 +25,21 @@ func Walk(path string, opts *WalkOptions) (*types.Node, error) {
 		log.Warningf("%s: %v", path, err)
 		return nil, nil
 	}
-	st, ok := f.Sys().(*syscall.Stat_t)
-	if !ok {
-		return nil, fmt.Errorf("could not cast %T to syscall.Stat_t", f.Sys())
+	//implements os dependent proper methods.
+	var diskUsage helper.DiskUsage = &helper.FsInfo{}
+	size, _ := diskUsage.GetSize(f)
+	if err != nil {
```

Comment:
the `err` value here is not updated, let's change `size, _ := diskUsage.GetSize(f)` to `size, err := diskUsage.GetSize(f)`